### DISCUSSION TITLE
Can anyone please help with the monitoring params that needs to be taken into account

### DISCUSSION THREAD

**Discussion by shikha-priyedarshi** [uber/cadence#5958] [Q&A]
Can anyone please help with the monitoring params that needs to be taken into account while using it with postgresql apart from the ones available on cadence  public website.

**Comment by demirkayaender** [comment]
We don't have PostgreSQL specific metrics since we don't use it internally but we have persistence metrics that we monitor and that will tell if there are issues with persistence. We can share them if you are interested in.
Postgres specific metrics would be more related to Postgres than Cadence.

**Reply by shikha-priyedarshi** [reply]
Hi Ender, Thank you so much for replying. 
I am looking forward for metrics such as what is the name of the workflow that is currently running, how many  workflow , activities are in queue or tasklist.
Also wanted to understand  how the max connections and idle connections which are mentioned in values.yaml is getting used.

**Reply by demirkayaender** [reply]
I see, here's where you can get them from. The ones you mentioned aren't related to any DB technology.
* To see the currently running workflows, please use cadence-web. You can find it here: https://www.npmjs.com/package/cadence-web or here: https://github.com/uber/cadence-web
* To see how many workflows you are starting per hour, per second etc, you can use metrics starting with `cadence-workflow-*` (emitted from the client/worker side) and `workflow_*` (server side). Note that, these won't give you the total open workflow count but rather number of workflows started, completed, continued as new, signaled with start. 
* To see the total number of open workflows, you can either use cadence-web or you'd need to query your visibility store. Depending on which plugin you used, it could be ElasticSearch or Postgres. For ES it would look something like below:

      {
        "query": {
            "bool": {
                "must_not": {
                    "exists": {
                        "field": "CloseTime"
                    }
                }
            }
        }
    }

* You can find the backlog from `task_count_per_tl` metric emitted from cadence-matching service. You can use `tasklisttype:decision` or `tasklisttype:activity` to filter only for activities and decisions.

We hope to release a Grafana template for all the metrics we track soon.

**Reply by shikha-priyedarshi** [reply]
Thanks a lot.   I am not using ES. It would be really helpful if grafana dashboard is released for these.
Also wanted to understand how the max connections and idle connections which are mentioned in values.yaml is getting used.

**Reply by shikha-priyedarshi** [reply]
Adding to this, can we use this task_count_per_tl metric  in grafana ?

**Reply by demirkayaender** [reply]
`task_count_per_tl` is already emitted, you should be able to query in Grafana.
Total WF count metric will soon come as a metric too but today, it's only in visibility store. You can use Postgres as your visibility store. Then you can just write a count query filtering open workflows.

**Reply by shikha-priyedarshi** [reply]
Ok cool. Please Help me understand the max open max idle and max connections of values.yaml for cadence.

**Reply by demirkayaender** [reply]
Which values.yaml are you referring to? Are those Kubernetes configs? If so, I'm not 100% sure; we are using an internal technology for this.

**Reply by demirkayaender** [reply]
Although we plan to release official Helm chart configs soon, currently those configs are community built.

**Reply by shikha-priyedarshi** [reply]
I was mentioning about this one: https://github.com/banzaicloud/banzai-charts/blob/master/cadence/values.yaml#L167-L179

**Reply by demirkayaender** [reply]
They are the Helm charts. Banzai cloud stopped maintaining them, the most current ones should be here: https://github.com/edmondop/cadence-helm-chart/

As I mentioned, these are community built at the moment and we will release official ones soon.

**Reply by shikha-priyedarshi** [reply]
Thank you so much. These all are really very valuable inputs.
Thanks a lot.

**Reply by shikha-priyedarshi** [reply]
Just one last one :          
         # maxConns: 20
          # maxIdleConns: 20
          # maxConnLifetime: "1h"
          
          How do these  getting utilized in your internal setup. are these per pod connections ?

**Reply by demirkayaender** [reply]
Sorry I'm not really familiar with this config at the moment. You might want to try and monitor. You can also try raising an issue in that repo.

**Reply by shikha-priyedarshi** [reply]
Sure not an issue. Thanks a lot for your help here.

**Reply by demirkayaender** [reply]
Closing this issue for now. Feel free to reopen if your problems persist.
